examples/domainmodel/src/cli/cli-util.ts

```diff
@@ -0,0 +1,44 @@
+/******************************************************************************
+ * Copyright 2021 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+import * as fs from 'fs';
+import colors from 'colors';
+import { DefaultDocumentValidator, Grammar, LangiumDocumentConfiguration, LangiumServices } from 'langium';
+
+export function extractGrammar(fileName: string, languageId: string, extensions: string[], services: LangiumServices): Grammar {
```

Comment:
Why does this return a `Grammar` if its then casted into a `DomainModel` either way? I would convert it into a generic method, simply returning `T`.

Reply:
`Grammar` extends `AstNode` and has more functionality, so if I return `T`, I lose it. Actually, I don't need this functionality now, for domain model CLI, but it's income in extracting common API.

Reply:
Right, we usually use a generic constraint for that in most of the Langium code which would look like this `extract<T extends AstNode>(...): T`. Though it is not necessary for this case.